src/main/java/seedu/address/model/project/DeadlineList.java

```diff
@@ -69,7 +69,12 @@ public void deleteDeadline(Integer i) {
      */
     public void markAsDone(Integer i) {
         requireNonNull(i);
-        this.deadlines.get(i).markAsDone();
+        CompletableDeadline deadline = deadlines.get(i);
+        deadline.markAsDone();
+
+        // Force observable list to update
+        this.deadlines.remove(deadline);
+        this.deadlines.add(i, deadline);
```

Comment:
Can the Set(i,deadline) method be used instead. This way the list would not get reordered unnecessarily.